lib/segment/src/index/field_index/full_text_index/inverted_index.rs

```diff
@@ -165,7 +180,7 @@ impl InvertedIndex {
             };
         }
         // Smallest posting is the largest possible cardinality
-        let smallest_posting = postings.iter().map(|posting| posting.len()).min().unwrap();
+        let smallest_posting = postings.iter().min().copied().unwrap();
```

Comment:
Also I think `rust-analyzer` have troubles to infer `postings_opt` type now. 

It shows me `impl Iterator<Item = Option<...>>` before the collect, but where another layer of option goes, assuming that `postings` is `Vec<usize>`

Reply:
Added `Vec<usize>` type